server/usercache.go

```diff
@@ -88,10 +88,14 @@ func (luc *ldapUserCache) Update() error {
 		userKeys := []ssh.PublicKey{}
 		for _, eachKey := range entry.GetAttributeValues("sshPublicKey") {
 			sshKeyBytes, _ := base64.StdEncoding.DecodeString(eachKey)
-			userSSHKey, err := ssh.ParsePublicKey(sshKeyBytes)
-			if err != nil {
-				log.Errorf("SSH key parsing for %s failed! This key will not be added into LDAP.", username)
-				continue
+			userSSHKey, err1 := ssh.ParsePublicKey(sshKeyBytes)
+			if err1 != nil {
+				userSSHKey2, _, _, _, err2 := ssh.ParseAuthorizedKey([]byte(eachKey))
```

Comment:
If you look at my example, note that I use "=" instead of ":=", that was probably your issue :)
:= will allocate new variables on the context of your if, which is not what you want in this case. This might help illustrate the difference: http://play.golang.org/p/b8y6SPl_lK

Thanks for helping out btw!